javaparser-core/src/main/java/com/github/javaparser/javadoc/description/JavadocInlineTag.java

```diff
@@ -97,8 +97,7 @@ public boolean equals(Object o) {
 
         JavadocInlineTag that = (JavadocInlineTag) o;
 
-        if (type != that.type) return false;
-        return content.equals(that.content);
+        return type == that.type && content.equals(that.content);
```

Comment:
If we allow for the Unknown type we should probably store the actual name of the tag used and use it also in the equals method and the toString method